src/segmenter/japanese.rs

```diff
@@ -0,0 +1,36 @@
+use crate::segmenter::Segmenter;
+use lindera::tokenizer::{Tokenizer as LinderaTokenizer, TokenizerConfig};
+use lindera_core::viterbi::{Mode, Penalty};
+use once_cell::sync::Lazy;
+
+/// Japanese specialized [`Segmenter`].
+///
+/// This Segmenter uses lindera internally to segment the provided text.
+pub struct JapaneseSegmenter;
+
+static LINDERA: Lazy<LinderaTokenizer> = Lazy::new(|| {
+    let config =
+        TokenizerConfig { mode: Mode::Decompose(Penalty::default()), ..TokenizerConfig::default() };
+    LinderaTokenizer::with_config(config).unwrap()
+});
+
+impl Segmenter for JapaneseSegmenter {
+    fn segment_str<'o>(&self, to_segment: &'o str) -> Box<dyn Iterator<Item = &'o str> + 'o> {
+        let segment_iterator = LINDERA.tokenize_str(to_segment).map(|vec| vec.into_iter()).unwrap();
+        Box::new(segment_iterator)
```

Comment:
```suggestion
        let segments = LINDERA.tokenize_str(to_segment).unwrap();
        Box::new(segments.into_iter())
```
This should work and I find it easier to read, what do you think? 😄 

About the unwrap I'll investigate later 🤔